contracts/MatchPayouts.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.7.5;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+/**
+ * @dev This contract allows for non-custodial Gitcoin Grants match payouts. It works as follows:
+ *  1. During a matching round, deploy a new instance of this contract
+ *  2. Once the round is complete, Gitcoin computes the final match amounts earned by each grant
+ *  3. Over the course of multiple transactions, the contract owner will set the payout mapping
+ *     stored in the `payouts` variable. This maps each grant receiving address to the match amount
+ *     owed, in DAI
+ *  4. Once this mapping has been set for each grant, the contract owner calls `finalize()`. This
+ *     sets `finalized` to `true`, and at this point the payout mapping can no longer be updated.
+ *  5. Funders review the payout mapping, and if they approve they transfer their funds to this
+ *     contract. This can be done with an ordinary transfer to this contract address.
+ *  6. Once all funds have been transferred, the contract owner calls `enablePayouts` which lets
+ *     grant owners withdraw their match payments
+ *  6. Grant owners can now call `withdraw()` to have their match payout sent to their address.
+ *     Anyone can call this method on behalf of a grant owner, which is useful if your Gitcoin
+ *     grants address cannot call contract methods.
+ * 
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *        WARNING: DO NOT SEND ANYTHING EXCEPT FOR DAI TO THIS CONTRACT OR IT WILL BE LOST!        *
```

Comment:
Should we consider adding an `onlyOwner` escape hatch that allows sending any ERC20 funds that *aren't* `dai` out to an arbitrary address.

Reply:
@owocki decided against this to minimize legal/custodial risk